src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/SqlManagerController.php

```diff
@@ -0,0 +1,337 @@
+                $this->addFlash('error', $error);
+            }
+        }
+
+        $params = [
+            'requestSqlForm' => $form->createView(),
+            'dbTableNames' => $dataProvider->getTables(),
+        ];
+
+        return $this->getTemplateParams($request, false) + $params;
+    }
+
+    /**
+     * Show Request SQL edit page
+     *
+     * @Template("@PrestaShop/Admin/Configure/AdvancedParameters/SqlManager/form.html.twig")
+     *
+     * @param int $id
+     * @param Request $request
+     *
+     * @return array|RedirectResponse
+     */
+    public function editAction($id, Request $request)
+    {
+        $formHandler = $this->get('prestashop.admin.request_sql.form_handler');
+        $dataProvider = $this->get('prestashop.adapter.sql_manager.request_sql_data_provider');
+
+        $form = $formHandler->getForm();
+        $form->setData(['request_sql' => $dataProvider->getRequestSql($id)]);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted()) {
+            $data = $form->getData();
+
+            if (!$errors = $formHandler->save($data)) {
+                $this->addFlash('success', $this->trans('Successful update.', 'Admin.Notifications.Success'));
+
+                return $this->redirectToRoute('admin_sql_manager');
+            }
+
+            foreach ($errors as $error) {
+                $this->addFlash('error', $error);
+            }
+        }
+
+        $params = [
+            'requestSqlForm' => $form->createView(),
+            'dbTableNames' => $dataProvider->getTables(),
+        ];
+
+        return $this->getTemplateParams($request) + $params;
+    }
+
+    /**
+     * Delete selected Request SQL
+     *
+     * @param int $id           ID of selected Request SQL
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function deleteAction($id, Request $request)
+    {
+        $legacyController = $request->attributes->get('_legacy_controller');
+
+        if ($this->isDemoModeEnabled()) {
+            $this->addFlash('error', $this->getDemoModeErrorMessage());
+
+            return $this->redirectToRoute('admin_sql_manager');
+        }
+
+        if ($this->authorizationLevel($legacyController) != PageVoter::LEVEL_DELETE) {
+            $this->addFlash('error', $this->trans('You do not have permission to delete this.', 'Admin.Notifications.Error'));
+
+            return $this->redirectToRoute('admin_sql_manager');
+        }
+
+        $requestSqlManager = $this->get('prestashop.adapter.sql_manager.request_sql_manager');
+        $requestSqlManager->delete($id);
+
+        $this->addFlash('success', $this->trans('Successful deletion', 'Admin.Notifications.Success'));
+
+        return $this->redirectToRoute('admin_sql_manager');
+    }
+
+    /**
+     * Get table columns data
+     *
+     * @param string $table     Database tabe name
+     *
+     * @return JsonResponse
+     */
+    public function tableColumnsAction($table)
```

Comment:
this action may be reused and could be part of CommonController

Reply:
it's not related to dynamic tables, not sure if needed.

Reply:
oh? the name of action is confusing to me then :p